packages/cli/src/bundle/metro.ts

```diff
@@ -32,10 +32,18 @@ export async function metroBundle(
 ): Promise<void> {
   info(`Bundling ${bundleConfig.platform}...`);
 
-  customizeMetroConfig(metroConfig, {
-    ...bundleConfig,
-    treeShake: !dev && bundleConfig.treeShake,
-  });
+  if (!dev && bundleConfig.treeShake) {
+    if (minify !== undefined) {
+      bundleConfig.treeShake =
+        typeof bundleConfig.treeShake === "object"
+          ? { ...bundleConfig.treeShake, minify }
+          : { minify };
```

Comment:
Let's avoid the copy if we can:
```suggestion
      if (typeof bundleConfig.treeShake === "object") {
        bundleConfig.treeShake.minify = minify;
      } else {
        bundleConfig.treeShake = { minify };
      }
```